include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -8874,6 +8874,18 @@ def err_bounds_type_annotation_lost_checking : Error<
   def err_expected_bounds : Error<
     "expression has no bounds">;
 
+  def err_expected_bounds_for_ptr_cast : Error<
+    "expression has no bounds"
+    ", cast to ptr<T> expects source to have bounds">;
+
+  def err_expected_bounds_for_assignment : Error<
+    "expression has no bounds"
+    ", right-hand side of assignment expected to have bounds because the left-hand side has bounds">;
+
+  def err_expected_bounds_for_initializer : Error<
+    "expression has no bounds"
+    ", an initializer expected to have bounds because variable declaration has bounds">;
```

Comment:
This error message is slightly grammatically incorrect.  I would reword it as "initializer expected to have bounds because the variable being declared has bounds."